websockets/websockets_test.go

```diff
@@ -315,8 +316,94 @@ func TestBinaryState(t *testing.T) {
 	`))
 	require.NoError(t, err)
 	logs := hook.Drain()
-	require.Len(t, logs, 1)
-	require.Contains(t, logs[0].Message, binarytypeWarning)
+	require.Len(t, logs, 0)
+}
+
+func TestBinaryType(t *testing.T) {
+	t.Parallel()
+
+	t.Run("default", func(t *testing.T) {
+		ts := newTestState(t)
+		logger, hook := testutils.NewLoggerWithHook(t, logrus.WarnLevel)
+		ts.runtime.VU.StateField.Logger = logger
+		_, err := ts.runtime.RunOnEventLoop(ts.tb.Replacer.Replace(`
+		var ws = new WebSocket("WSBIN_URL/ws-echo")
+		ws.addEventListener("open", () => {
+			const sent = new Uint8Array([164,41]).buffer
+			ws.send(sent)
+			ws.onmessage = (e) => {
+				//if (!(e.data instanceof Blob)) {
+				//	throw new Error("Wrong event.data type; expected: Blob, got: "+ typeof e.data)
+				//}
+
+				if (sent.byteLength !== e.data.arrayBuffer().byteLength) {
+					throw new Error("The data received isn't equal to the data sent")
+				}
+
+				ws.close()
+			}
+		})
+	`))
+		require.NoError(t, err)
+		logs := hook.Drain()
+		require.Len(t, logs, 0)
+	})
+
+	t.Run("blob", func(t *testing.T) {
+		ts := newTestState(t)
+		logger, hook := testutils.NewLoggerWithHook(t, logrus.WarnLevel)
+		ts.runtime.VU.StateField.Logger = logger
+		_, err := ts.runtime.RunOnEventLoop(ts.tb.Replacer.Replace(`
+		var ws = new WebSocket("WSBIN_URL/ws-echo")
+		ws.binaryType = "blob"
+		ws.addEventListener("open", () => {
+			const sent = new Uint8Array([164,41]).buffer
+			ws.send(sent)
+			ws.onmessage = (e) => {
+				//if (!(e.data instanceof Blob)) {
+				//	throw new Error("Wrong event.data type; expected: Blob, got: "+ typeof e.data)
+				//}
```

Comment:
I would argue this is blocking - this is *especially* blocking if we want to move this to the core.

JS objects that do not work with instanceof aren't a good thing. K6 in general doesn't care about this as most of the objects we have - are not meant to be constructed from outside, but Blob isn't one of those. Another example of htis will be http.File.

Reply:
Probably the easiest way to get around this in this case will be https://github.com/dop251/goja/issues/10#issuecomment-274339012 as otherwise the alternative is to create a prototype and reuse it.

Which unfortunately involves a lot more code and all of the examples I can find in goja/sobek are ... way too complicated

Reply:
I decided to go with what's suggested here: https://github.com/dop251/goja/issues/529